### PR TITLE
Added documentation for the new FileStreamSecurityValidator

### PR DESCRIPTION
**Pull request by Migaroez**
## Description

Added docs for the new FileStreamSecurityValidator in all affected versions.

## Type of suggestion

* [ ] Typo/grammar fix
* [ ] Updated outdated content
* [x] New content
* [ ] Updates related to a new version
* [ ] Other

## Product & version (if relevant)
CMS v10 and upwards

## Deadline (if relevant)
Should be published together wit the release that contains this PR
https://github.com/umbraco/Umbraco-CMS/pull/14657, minimum version numbers need to be updated before publishing

### CODE REVIEW

File: 10/umbraco-cms/reference/security/serverside-file-validation.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+### Implementing a FileStreamSecurityValidator
+The `IFileStreamSecurityAnalyzer` consists of 2 methods:
+- `FileContentMatchesFileType`: This method should return true if your validator believes it is responsible for checking this file
+- `IsConsideredSafe`: This method should return false if the analyzer finds reason not to trust the file
+
+### Example FileStreamSecurityValidator
+The following class shows how one could potentially guard against Cross-site scripting(XSS) vulnerabilities in an svg file.
+
+```
```
Comment (alina-tincas) [inline 10/umbraco-cms/reference/security/serverside-file-validation.md:21]: as it is csharp it makes the format to match it 😁 
```suggestion
```csharp
```

**Review by alina-tincas — CHANGES_REQUESTED** [review]
Hi @Migaroez 👋 

Thank you for the PR! 🙌 

In order for the pages to show up in the documentation navigation, can you please add the reference to those pages in the[ SUMMARY.md ](https://docs.umbraco.com/welcome/contribute/pull-request#step-1-creating-a-fork) as well? 😁 

Also, once you would like these changes merged in, please let us know 👍

**Comment by alina-tincas** [comment]
Hi @Migaroez 👋 
I was wondering if you have seen my previous suggestions and if you are able to make the mentioned changes?

Also, please let me know once all those changes should be merged in 👍

**Comment by Migaroez** [comment]
Versions to add to this docs

[12.2.0-rc](https://www.nuget.org/packages/Umbraco.Cms/12.2.0-rc) - Since September 7th - 2023
[11.5.0](https://www.nuget.org/packages/Umbraco.Cms/11.5.0) - Since September 7th - 2023
[10.7.0](https://www.nuget.org/packages/Umbraco.Cms/10.7.0) - Since September 7th - 2023
[8.18.9](https://www.nuget.org/packages/UmbracoCms/8.18.9) - Since August 28th - 2023
[7.15.11](https://www.nuget.org/packages/UmbracoCms/7.15.11) - Since September 5th - 2023

**Pull request merged**